### ISSUE TITLE
Accessibility: Visual label relies on placeholder text [All Products]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52135]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The visual labels for these input fields rely only on the placeholder text. Thus, the label is not available for all users. After a user types in the input, the placeholder text disappears so, for example, if the user needs to review the data there is no way to confirm the info is placed in the correct input, etc. The lack of visible and descriptive labels might cause difficulties for all users when trying to enter information—but particularly for those with cognitive, language, and learning disabilities.

![Image](https://github.com/user-attachments/assets/b8aca8fa-29c9-4bf5-8dfd-3619c1f81bbe)

### Component or Page: 

Component: All Products Block     

### Name affected element:

Orders Product by combobox

### Relevant code snippets:

`<div class="wc-block-sort-select wc-block-components-sort-select wc-block-product-sort-select wc-block-components-product-sort-select"><label class="wc-block-sort-select__label wc-block-components-sort-select__label" for="wc-block-components-sort-select__select-0"><span aria-hidden="true"></span><span class="screen-reader-text">Order products by</span></label><select id="wc-block-components-sort-select__select-0" class="wc-block-sort-select__select wc-block-components-sort-select__select"><option value="menu_order">Default sorting</option><option value="popularity">Popularity</option><option value="rating">Average rating</option><option value="date">Latest</option><option value="price">Price: low to high</option><option value="price-desc">Price: high to low</option></select></div> `

### Recommended fix:

Provide a label for the input. Add a <label> tag and use a "for" attribute to link the label to the input. Ensure the label is clear and descriptive.

Add "Order by" to the left of the flied as has been done on the reviews block.

**Comment by gigitux** [comment]
We are going to soft-deprecate the [`All Products` block](https://developer.woocommerce.com/2024/11/11/product-collection-block-new-js-events-and-soft-deprecation-of-product-grid-blocks/#:~:text=Upcoming%20Soft%20Deprecation%20of%20Product%20Grid%20Blocks%3A). I think that we can close this issue: WDYT?

**Comment by vikrampm1** [comment]
cc @amberhinds ^

**Comment by amberhinds** [comment]
I'm not 100% sure what "soft deprecate" the block means. Do you plan to remove it? Or will it just no longer be supported? Will there be a notice in the editor advising people not to use the block?

If you don't want to fix the block and are planning to remove it in the future, this is how we will handle it on the the VPAT/ACR:

- The relevant success criterion will have "partially supports" rather than "supports."
- In the notes column for that success criterion, we will note that this block does not meet the criterion, but it is deprecated as of version X of the software, and users are advised not to use it.

That is an acceptable approach; it's more of an internal question about whether Woo is OK with the "partially supports" versus "supports" in the VPAT for the relevant criterion. 

From a marketing standpoint, I doubt that will be an issue if it's just this one block, it's clearly explained in the notes, and there's an alternative for users. 

@jamesckemp do you have an opinion on how you want to approach issues on deprecated components?

**Comment by gigitux** [comment]
> I'm not 100% sure what "soft deprecate" the block means. Do you plan to remove it? Or will it just no longer be supported? Will there be a notice in the editor advising people not to use the block?


[We just published a dedicated post](https://developer.woocommerce.com/2024/11/19/product-collection-block-the-new-default/). TLDR: New user can't add this block because it is hide from the inserter, but the existent instances will be still full working.

**Comment by amberhinds** [comment]
Thanks, that's helpful. It is still an issue for sites that already have it inserted, but it is helpful to know that it can't be inserted on new sites.

**Comment by MajoMaldonado** [comment]
We are currently testing this issue, but we are not seeing any changes on the front-end, nor can we find a way to toggle the functionality to display the label on the back-end. Could you please assist us so we can confirm whether the issue has been resolved?
cc: @amberhinds

**Comment by amberhinds** [comment]
@Manussakis @imanish003 @jeffpaul This one also does not appear to be fixed, unless we're missing something. The linked PR does not appear to have corrected the issue.

Reading up the comments, I am wondering if this issue was supposed to be closed as "Not going to fix"? Can you please clarify if this is believed to be fixed and, if so, how we can get the visible label to show?

**Comment by Manussakis** [comment]
@amberhinds @MajoMaldonado Apologies, for some reason I associated this issue with the https://github.com/woocommerce/woocommerce/pull/52838 PR, but this PR only fixes the label issue for the Catalog Sorting block (https://github.com/woocommerce/woocommerce/issues/51930)

@gigitux Considering [this comment you left](https://github.com/woocommerce/woocommerce/issues/52135#issuecomment-2485481678), please let me know if the Woo team wants to invest time on this and I can reopen this issue if needed.

Thank you all!

**Comment by amberhinds** [comment]
Thanks. We'll leave this issue as not fixed on our audit report for now.

**Comment by amberhinds** [comment]
@Manussakis @jamesckemp @vikrampm1 @gigitux - it looks like this issue is still waiting on direction. It was incorrectly linked with a PR that did not fix it. Should this be reopened or closed as not fixing?

**Comment by gigitux** [comment]
`All Products` block is a deprecated block. I think that we can keep it closed as not fixing.

**Comment by Manussakis** [comment]
Tagging @vikrampm1 for visibility of the @gigitux comment above.
